clients/application-kafka/src/main/java/org/eclipse/hono/application/client/kafka/impl/KafkaBasedAsyncCommandClient.java

```diff
@@ -0,0 +1,160 @@
+     */
+    public static Future<AsyncCommandClient> create(
+            final String tenantId,
+            final KafkaProducerFactory<String, Buffer> producerFactory,
+            final KafkaProducerConfigProperties config,
+            final Tracer tracer) {
+        final KafkaBasedAsyncCommandClient asyncCommandClient = new KafkaBasedAsyncCommandClient(tenantId,
+                producerFactory, config, tracer);
+        return asyncCommandClient.start().map(asyncCommandClient);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * <p>
+     * The replyId is not used in the Kafka based implementation. It can be set to {@code null}.
+     * If set it will be ignored.
+     *
+     * @throws NullPointerException if deviceId, command or correlationId is {@code null}.
+     */
+    @Override
+    public Future<Void> sendAsyncCommand(
+            final String deviceId,
+            final String command,
+            final Buffer data,
+            final String correlationId,
+            final String replyId) {
+        return sendAsyncCommand(deviceId, command, null, data, correlationId, replyId, null);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * <p>
+     * The replyId is not used in the Kafka based implementation. It can be set to {@code null}.
+     * If set it will be ignored.
+     *
+     * @throws NullPointerException if deviceId, command or correlationId is {@code null}.
+     */
+    @Override
+    public Future<Void> sendAsyncCommand(
+            final String deviceId,
+            final String command,
+            final String contentType,
+            final Buffer data,
+            final String correlationId,
+            final String replyId,
+            final Map<String, Object> properties) {
+        return sendAsyncCommand(deviceId, command, contentType, data, correlationId, replyId, properties,
+                NoopSpan.INSTANCE.context());
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * <p>
+     * The replyId is not used in the Kafka based implementation. It can be set to {@code null}.
+     * If set it will be ignored.
+     *
+     * @throws NullPointerException if deviceId, command or correlationId is {@code null}.
+     */
+    @Override
+    public Future<Void> sendAsyncCommand(
+            final String deviceId,
+            final String command,
+            final String contentType,
+            final Buffer data,
+            final String correlationId,
+            final String replyId,
+            final Map<String, Object> properties,
+            final SpanContext context) {
+        Objects.requireNonNull(deviceId);
+        Objects.requireNonNull(command);
+        Objects.requireNonNull(correlationId);
+
+        final List<KafkaHeader> headers = getHeaders(deviceId, command, contentType, correlationId);
+        return sendAndWaitForOutcome(topic.toString(), tenantId, deviceId, data, headers, context);
```

Comment:
In the above place since we already have a map of properties, it makes sense to add the additional properties to the Map pass the `Map` as an argument.

> TBH I still think the existence of this overloaded method is not justified.

In command router, we forward commands to the internal Kafka topic and there it is helpful to pass a list of headers from the `KafkaConsumerRecord`. Otherwise the headers need to be converted to a map of properties, which will finally get converted to a list of headers.